samples/client/petstore/typescript-inversify/api/store.service.ts

```diff
@@ -0,0 +1,121 @@
+        @inject("IAPIConfiguration") private APIConfiguration: IAPIConfiguration ) {
+        if(this.APIConfiguration.basePath)
+            this.basePath = this.APIConfiguration.basePath;
+    }
+
+
+    /**
+     * Delete purchase order by ID
+     * For valid response try integer IDs with value &lt; 1000. Anything above 1000 or nonintegers will generate API errors
+     * @param orderId ID of the order that needs to be deleted
+     
+     */
+
+    public deleteOrder(orderId: string, headers: Dictionary<string> = {}): Promise<any> {
+        if (!orderId){
+            throw new Error('Required parameter orderId was null or undefined when calling deleteOrder.');
+        }
+
+        headers['Accept'] = 'application/xml';
+
+
+        return this.httpClient.delete(`${this.basePath}/store/order/${encodeURIComponent(String(orderId))}`, headers)
+                    .map(httpResponse => <any>(httpResponse.response))
+                    .toPromise();
+    }
+
+
+    /**
+     * Returns pet inventories by status
+     * Returns a map of status codes to quantities
+     
+     */
+
+    public getInventory(headers: Dictionary<string> = {}): Promise<{ [key: string]: number; }> {
+
+        // authentication (api_key) required
+        if (this.APIConfiguration.apiKeys["api_key"]) {
+            headers['api_key'] = this.APIConfiguration.apiKeys["api_key"];
+        }
+        headers['Accept'] = 'application/json';
+
+
+        return this.httpClient.get(`${this.basePath}/store/inventory`, headers)
+                    .map(httpResponse => <{ [key: string]: number; }>(httpResponse.response))
+                    .toPromise();
+    }
+
+
+    /**
+     * Find purchase order by ID
+     * For valid response try integer IDs with value &lt;&#x3D; 5 or &gt; 10. Other values will generated exceptions
+     * @param orderId ID of pet that needs to be fetched
+     
+     */
+
+    public getOrderById(orderId: number, headers: Dictionary<string> = {}): Promise<Order> {
+        if (!orderId){
+            throw new Error('Required parameter orderId was null or undefined when calling getOrderById.');
+        }
+
+        headers['Accept'] = 'application/xml';
+
+
+        return this.httpClient.get(`${this.basePath}/store/order/${encodeURIComponent(String(orderId))}`, headers)
+                    .map(httpResponse => <Order>(httpResponse.response))
+                    .toPromise();
+    }
+
+
+    /**
+     * Place an order for a pet
+     * 
+     * @param body order placed for purchasing the pet
+     
+     */
+
+    public placeOrder(body: Order, headers: Dictionary<string> = {}): Promise<Order> {
+        if (!body){
+            throw new Error('Required parameter body was null or undefined when calling placeOrder.');
+        }
+
+        headers['Accept'] = 'application/xml';
+        headers['Content-Type'] = 'application/json';
+
+
+        return this.httpClient.post(`${this.basePath}/store/order`, body , headers)
```

Comment:
are you sure you want to convert all response observables to promises?
returning them as observables would make it convenient to chain with `RxJS`, similar to the Angular `HttpClient`.
As a drawback, you always need to subscribe to the API call observable, otherwise the request is not performed.

Reply:
On this subject I was undecided, because the observable returned by the HttpClient complete immediately and returns only a value (as every Promise). But you have reason that returing a observable would make convenient to chain with `RxJS`, but you can do it with the operator fromPromise if necessary... I'm very undecided about this.

Reply:
I would suggest to adapt the interface of the Angular HttpClient with Observables. 
Anyone else got an opinion on this?

Reply:
I would add a configuration variable that allow the user to decide if use Promise or Observable. What about this solution?

Reply:
@macjohnny i've added the `usePromise` config's variable.